ethereum_history_api/circuits/lib/src/serde.nr

```diff
@@ -111,3 +102,75 @@ impl Serde<STORAGE_BLOCK_LEN> for StorageWithinBlock<1> {
         }
     }
 }
+
+global RECEIPT_MAX_DEPTH_NO_LEAF = 6;
+global RECEIPT_NODES_LEN = 3192; // MAX_NODE_LEN * MAX_DEPTH_NO_LEAF;
+
+pub fn serialize_nodes(nodes: [Node; RECEIPT_MAX_DEPTH_NO_LEAF]) -> [Field; RECEIPT_NODES_LEN] {
+    let mut data: BoundedVec<Field, RECEIPT_NODES_LEN> = BoundedVec::new();
+    for i in 0..RECEIPT_MAX_DEPTH_NO_LEAF {
+        data.extend_from_array(nodes[i].serialize());
+    }
+    data.storage
+}
+
+pub fn deserialize_nodes(data: [Field; RECEIPT_NODES_LEN]) -> [Node; RECEIPT_MAX_DEPTH_NO_LEAF] {
+    let mut result: Fragment<RECEIPT_MAX_DEPTH_NO_LEAF, Node> = Fragment::empty();
+    let mut fragment = Fragment::from_array(data);
+    for _ in 0..RECEIPT_MAX_DEPTH_NO_LEAF {
+        let node: Node = fragment.pop_front_array().deserialize();
+        result.push_back(node);
+    }
+    result.data
+}
+
+global RECEIPT_MAX_LEAF_LEN_M = 1011;
+global RECEIPT_PROOF_LEN_M = RECEIPT_MAX_DEPTH_NO_LEAF * MAX_NODE_LEN + RECEIPT_MAX_LEAF_LEN_M + 1;
+
+impl Serde<RECEIPT_PROOF_LEN_M> for Proof<RECEIPT_MAX_DEPTH_NO_LEAF, RECEIPT_MAX_LEAF_LEN_M> {
+    fn serialize(self) -> [Field; RECEIPT_PROOF_LEN_M] {
+        let mut data: BoundedVec<Field, RECEIPT_PROOF_LEN_M> = BoundedVec::new();
+        data.extend_from_array(serialize_nodes(self.nodes));
+        data.extend_from_array(self.leaf.serialize());
+        data.push(self.depth as Field);
+        data.storage
+    }
+
+    fn deserialize(data: [Field; RECEIPT_PROOF_LEN_M]) -> Self {
+        let mut fragment = Fragment::from_array(data);
+        let nodes: [Node; RECEIPT_MAX_DEPTH_NO_LEAF] = deserialize_nodes(fragment.pop_front_array());
+        let leaf = fragment.pop_front_array().deserialize();
+        let depth = fragment.pop_front() as u64;
+        Proof {
+            nodes,
+            leaf,
+            depth,
+        }
+    }
+}
+
+global RECEIPT_MAX_PREFIXED_KEY_NIBBLE_LEN = 8;
+global RECEIPT_MAX_VALUE_LEN_M = 1000;
```

Comment:
Seems like some constants here are related to receipt in general and not serialization of the receipt. Those should not live here but in receipt.nr

Reply:
Moved